### ISSUE TITLE
NullPointerException: Cannot invoke "com.axway.apim.api.model.APIMethod.getName()"

### ISSUE THREAD

**Issue by renankaic** [Axway-API-Management-Plus/apim-cli#512]
### APIM-CLI version

1.14.6

### API-Management version

7.7.0.20230228

### Bug description

Hello.
I'm testing the new 1.14.6 apim cli version on my API Gateway Feb 2023.
But, I'm not being able to export an existing API from it due to the following error:
```
PS C:\Users\[my-user]\Desktop\tests> apim api get -s prod -id 38c1ad9f-596f-47e2-bbe3-c375df04aeea -useFEAPIDefinition --o json
2024-10-18 11:23:05,171 [APIManagerCLI] INFO : API-Manager CLI: 1.14.6
2024-10-18 11:23:05,175 [APIManagerCLI] INFO : Module: API - E X P O R T / U T I L S  (1.14.6)
2024-10-18 11:23:08,383 [APIManagerAdapter] INFO : Successfully connected to API-Manager (7.7.20230228) on: https://myapigatway.com:8075
2024-10-18 11:23:11,571 [APIManagerAPIAdapter] INFO : Loading application quotas for 475 subscribed applications.
2024-10-18 11:23:13,017 [APIExportApp] INFO : 1 API(s) selected.
2024-10-18 11:23:13,018 [JsonAPIExporter] INFO : Exporting API and configuration as JSON format
2024-10-18 11:23:13,090 [APIManagerAPIMethodAdapter] WARN : No operation found with ID: GET /GetLead for API: 38c1ad9f-596f-47e2-bbe3-c375df04aeea
2024-10-18 11:23:13,094 [APIExportApp] ERROR: Can't create API-Configuration file for API: 'My API - v3 - PRD' exposed on path: '/integration/v3/bdc'.
com.axway.apim.lib.error.AppException: Can't create API-Configuration file for API: 'My API - v3 - PRD' exposed on path: '/integration/v3/bdc'.
        at com.axway.apim.api.export.impl.ExportHelper.writeContent(ExportHelper.java:173)
        at com.axway.apim.api.export.impl.ExportHelper.saveAPILocally(ExportHelper.java:102)
        at com.axway.apim.api.export.impl.JsonAPIExporter.execute(JsonAPIExporter.java:32)
        at com.axway.apim.APIExportApp.execute(APIExportApp.java:184)
        at com.axway.apim.APIExportApp.exportAPI(APIExportApp.java:47)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.axway.apim.cli.APIManagerCLI.run(APIManagerCLI.java:130)
        at com.axway.apim.cli.APIManagerCLI.main(APIManagerCLI.java:72)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Cannot invoke "com.axway.apim.api.model.APIMethod.getName()" because the return value of "com.axway.apim.adapter.apis.APIManagerAPIMethodAdapter.getMethodForId(String, String)" is null (through reference chain: com.axway.apim.api.export.ExportAPI["applicationQuota"])
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:402)
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:361)
        at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:323)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:780)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:479)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:318)
        at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4719)
        at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:3906)
        at com.axway.apim.api.export.impl.ExportHelper.writeContent(ExportHelper.java:170)
        ... 10 more
Caused by: java.lang.NullPointerException: Cannot invoke "com.axway.apim.api.model.APIMethod.getName()" because the return value of "com.axway.apim.adapter.apis.APIManagerAPIMethodAdapter.getMethodForId(String, String)" is null
        at com.axway.apim.api.export.ExportAPI.translateMethodIds(ExportAPI.java:269)
        at com.axway.apim.api.export.ExportAPI.getApplicationQuota(ExportAPI.java:257)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:772)
        ... 16 more
PS C:\Users\[my-user]\Desktop\tests>
```

However, the same command works with 1.14.5 version as you can see below:
```
PS C:\Users\[my-user]\Desktop\tests> apim api get -s prod -id 38c1ad9f-596f-47e2-bbe3-c375df04aeea -useFEAPIDefinition --o json 
2024-10-18 11:18:25,533 [APIManagerCLI] INFO : API-Manager CLI: 1.14.5
2024-10-18 11:18:25,536 [APIManagerCLI] INFO : Module: API - E X P O R T / U T I L S  (1.14.5)
2024-10-18 11:18:28,466 [APIManagerAdapter] INFO : Successfully connected to API-Manager (7.7.20230228) on: https://myapigateway.com:8075
2024-10-18 11:18:29,068 [APIManagerAPIMethodAdapter] WARN : No operation found with ID: 38c1ad9f-596f-47e2-bbe3-c375df04aeea for API: 38c1ad9f-596f-47e2-bbe3-c375df04aeea
2024-10-18 11:18:31,549 [APIManagerAPIAdapter] INFO : Loading application quotas for 475 subscribed applications.
2024-10-18 11:18:32,894 [APIExportApp] INFO : 1 API(s) selected.
2024-10-18 11:18:32,895 [JsonAPIExporter] INFO : Exporting API and configuration as JSON format
2024-10-18 11:18:32,946 [ExportHelper] INFO : Successfully exported API: My API - v3 - PRD into folder: C:\Users\[my-user]\Desktop\tests\integration-v3-bdc
PS C:\Users\[my-user]\Desktop\tests>
```

### Steps to reproduce

1. Download and extract APIM CLI 1.14.6 version zip and configure the "AXWAY_APIM_CLI_HOME" env variable to the extracted folder
2. Try to export the frontend API with the command `apim api get -s prod -id [your-frontendapi-id] -useFEAPIDefinition --o json` that, for an unknown reason, receives the warn `[APIManagerAPIMethodAdapter] WARN : No operation found with ID:` when trying to export it
3. Receive the NullPointerException
4. Try again the step no. 2 but with 1.14.5 version of APIM CLI and see that the api export works flawlessly

### Relevant log output

```shell
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Cannot invoke "com.axway.apim.api.model.APIMethod.getName()" because the return value of "com.axway.apim.adapter.apis.APIManagerAPIMethodAdapter.getMethodForId(String, String)" is null (through reference chain: com.axway.apim.api.export.ExportAPI["applicationQuota"])
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:402)
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:361)
        at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:323)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:780)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:479)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:318)
        at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4719)
        at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:3906)
        at com.axway.apim.api.export.impl.ExportHelper.writeContent(ExportHelper.java:170)
        ... 10 more
Caused by: java.lang.NullPointerException: Cannot invoke "com.axway.apim.api.model.APIMethod.getName()" because the return value of "com.axway.apim.adapter.apis.APIManagerAPIMethodAdapter.getMethodForId(String, String)" is null
        at com.axway.apim.api.export.ExportAPI.translateMethodIds(ExportAPI.java:269)
        at com.axway.apim.api.export.ExportAPI.getApplicationQuota(ExportAPI.java:257)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:772)
        ... 16 more
```

**Comment by renankaic** [comment]
I've also checked the API swagger and it seems that all paths `operationId` are defined properly.

**Comment by rathnapandi** [comment]
Hi @renankaic, 

Can you share the api-config.json and spec from 1.14.5 export. if it has sensitive info, share it via my email (rnatarajan@axway.com)

**Comment by renankaic** [comment]
> Hi @renankaic,
> 
> Can you share the api-config.json and spec from 1.14.5 export. if it has sensitive info, share it via my email ([rnatarajan@axway.com](mailto:rnatarajan@axway.com))

Hi @rathnapandi , as you requested, I sent you by email

Let me know if you need any other information.

**Comment by rathnapandi** [comment]
[axway-apimcli-1.14.7-SNAPSHOT.tar.gz](https://github.com/user-attachments/files/17438594/axway-apimcli-1.14.7-SNAPSHOT.tar.gz)
Please use the attached binary for your testing.

**Comment by renankaic** [comment]
> [axway-apimcli-1.14.7-SNAPSHOT.tar.gz](https://github.com/user-attachments/files/17438594/axway-apimcli-1.14.7-SNAPSHOT.tar.gz) Please use the attached binary for your testing.

It worked!
See the last lines of the log:
```
2024-10-21 11:14:01,205 [APIExportApp] INFO : 1 API(s) selected.
2024-10-21 11:14:01,206 [YamlAPIExporter] INFO : Export API and configuration as Yaml format
2024-10-21 11:14:01,216 [ExportHelper] DEBUG: Going to export API: Siebel - BDC - v3 - PRD into folder: C:\Users\b1400143\Desktop\tests\integration-v3-bdc
2024-10-21 11:14:01,338 [ExportHelper] INFO : Successfully exported API: Siebel - BDC - v3 - PRD into folder: C:\Users\b1400143\Desktop\tests\integration-v3-bdc
2024-10-21 11:14:01,340 [APIManagerAdapter] DEBUG: Closing cache ...
2024-10-21 11:14:01,621 [RestAPICall] DEBUG: Http verb:DELETE and URI: https://bp1haebanmprd.hyundai-brasil.com:8075/api/portal/v1.4/login
```

Thank you very much @rathnapandi !
Can I already use this one you provided in our PRD environment or should I wait for the new release?

**Comment by rathnapandi** [comment]
Hi @renankaic,

please test all your use cases  and do let me know if you see any other issues. I will plan a release sometime next week.

**Comment by renankaic** [comment]
> Hi @renankaic,
> 
> please test all your use cases and do let me know if you see any other issues. I will plan a release sometime next week.

Hi @rathnapandi 
Most of tests in our use cases worked, except the backend certificate override as I sent you via email.

Please let me know if I did something wrong.

Thank you again.

**Comment by rathnapandi** [comment]
Fixed in 1.14.7
